Tests/GRPCHTTP2CoreTests/Client/Connection/LoadBalancers/RoundRobinLoadBalancerTests.swift

```diff
@@ -0,0 +1,440 @@
+        // for each.
+        let endpoints = context.servers.map { Endpoint(addresses: [$0.address]) }
+        context.loadBalancer.updateAddresses(endpoints)
+
+      case .connectivityStateChanged(.ready):
+        // Poll until each server has one connected client.
+        try await XCTPoll(every: .milliseconds(10)) {
+          context.servers.allSatisfy { server, _ in server.clients.count == 1 }
+        }
+
+        // Close to end the test.
+        context.loadBalancer.close()
+
+      default:
+        ()
+      }
+    } verifyEvents: { events in
+      let expected: [LoadBalancerEvent] = [
+        .connectivityStateChanged(.idle),
+        .connectivityStateChanged(.connecting),
+        .connectivityStateChanged(.ready),
+        .connectivityStateChanged(.shutdown),
+      ]
+      XCTAssertEqual(events, expected)
+    }
+  }
+
+  func testSubchannelsArePickedEvenly() async throws {
+    try await RoundRobinLoadBalancerTest.run(servers: 3, connector: .posix()) { context, event in
+      switch event {
+      case .connectivityStateChanged(.idle):
+        // Update the addresses for the load balancer, this will trigger subchannels to be created
+        // for each.
+        let endpoints = context.servers.map { Endpoint(addresses: [$0.address]) }
+        context.loadBalancer.updateAddresses(endpoints)
+
+      case .connectivityStateChanged(.ready):
+        // Subchannel is ready. This happens when any subchannel becomes ready. Loop until
+        // we can pick three distinct subchannels.
+        try await XCTPoll(every: .milliseconds(10)) {
+          var subchannelIDs = Set<SubchannelID>()
+          for _ in 0 ..< 3 {
+            let subchannel = try XCTUnwrap(context.loadBalancer.pickSubchannel())
+            subchannelIDs.insert(subchannel.id)
+          }
+          return subchannelIDs.count == 3
+        }
+
+        // Now that all are ready, load should be distributed evenly among them.
+        var counts = [SubchannelID: Int]()
+
+        for round in 1 ... 10 {
+          for _ in 1 ... 3 {
+            if let subchannel = context.loadBalancer.pickSubchannel() {
+              counts[subchannel.id, default: 0] += 1
+            } else {
+              XCTFail("Didn't pick subchannel from ready load balancer")
+            }
+          }
+
+          XCTAssertEqual(counts.count, 3, "\(counts)")
+          XCTAssert(counts.values.allSatisfy({ $0 == round }), "\(counts)")
+        }
+
+        // Close to finish the test.
+        context.loadBalancer.close()
+
+      default:
+        ()
+      }
+    } verifyEvents: { events in
+      let expected: [LoadBalancerEvent] = [
+        .connectivityStateChanged(.idle),
+        .connectivityStateChanged(.connecting),
+        .connectivityStateChanged(.ready),
+        .connectivityStateChanged(.shutdown),
+      ]
+      XCTAssertEqual(events, expected)
+    }
+  }
+
+  func testAddressUpdatesAreHandledGracefully() async throws {
```

Comment:
I think we're missing a test case where the updated endpoints exactly match existing endpoints. I suppose in this case we don't want to see a state update event.